apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -0,0 +1,128 @@
+import { ProcessSubscriberCommand } from './process-subscriber.command';
+import { matchMessageWithFilters } from '../trigger-event/message-filter.matcher';
+import { SendMessage } from '../send-message/send-message.usecase';
+import { SendMessageCommand } from '../send-message/send-message.command';
+import { SubscriberEntity } from '../../../../../../../libs/dal/src/repositories/subscriber/subscriber.entity';
+
+@Injectable()
+export class ProcessSubscriber {
+  constructor(
+    private subscriberRepository: SubscriberRepository,
+    private notificationRepository: NotificationRepository,
+    private createSubscriberUsecase: CreateSubscriber,
+    private createLogUsecase: CreateLog,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private sendMessage: SendMessage
+  ) {}
+
+  public async execute(command: ProcessSubscriberCommand) {
+    const template = await this.notificationTemplateRepository.findById(command.templateId, command.organizationId);
+
+    let notification;
+    let subscriber: SubscriberEntity;
+    try {
+      subscriber = await this.getSubscriber(command, template._id);
+      notification = await this.createNotification(command, template._id, subscriber.subscriberId);
+    } catch (e) {
+      return {
+        status: 'subscriber_not_found',
+      };
+    }
+
+    const steps = matchMessageWithFilters(template.steps, command.payload);
+    for (const step of steps) {
+      await this.sendMessage.execute(
+        SendMessageCommand.create({
+          identifier: command.identifier,
+          payload: command.payload,
+          step,
+          transactionId: command.identifier,
+          notificationID: notification._id,
+          environmentId: command.environmentId,
+          organizationId: command.organizationId,
+          userId: command.userId,
+          subscriberId: subscriber._id,
+        })
+      );
+    }
+
+    await this.createLogUsecase.execute(
+      CreateLogCommand.create({
+        transactionId: command.transactionId,
+        status: LogStatusEnum.INFO,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        notificationId: notification._id,
+        text: 'Request processed',
+        userId: command.userId,
+        subscriberId: subscriber._id,
+        code: LogCodeEnum.TRIGGER_PROCESSED,
+        templateId: notification._templateId,
+      })
+    );
+
+    return {
+      status: 'success',
+    };
+  }
+
+  private async getSubscriber(command: ProcessSubscriberCommand, templateId: string): Promise<SubscriberEntity> {
+    const subscriberPayload = command.to;
+    const subscriber = await this.subscriberRepository.findOne({
+      _environmentId: command.environmentId,
+      subscriberId: subscriberPayload.subscriberId,
+    });
+
+    if (subscriber) {
+      return subscriber;
+    }
+    if (subscriberPayload.email || subscriberPayload.phone) {
```

Comment:
As I mentioned in our brief maybe it is worth removing this condition because, for example, an in-app user must add an e-mail or phone in order to create the subscriber, while he may not have any use in them.
Well to tell the truth I am not sure it related to this PR !, just thought it was worth mentioning :)